### PR TITLE
Add unit test infrastructure to the platform

### PR DESCRIPTION
**Pull request by chelsea-tnc**
Background: This is part 1 of the work to prevent users from creating a duplicate "Archive" section. Before we add this functionality, let's set up some unit test infrastructure to ensure we're building this in a way that scales. 

What this PR does:
- Uses the latest version of Jest and adds new dependencies
- Includes instructions for running tests
- Includes a sample test that passes
- Fixes an accessibility bug: the form label was not explicitly assigned to the input element in TinyInputField

What this PR does not do:
- Fix pre-existing tests in hasuraQueriesAndMutations.js. I fixed a minor syntax error, but these tests are failing with the following error: `[{"extensions": {"code": "unexpected", "path": "$"}, "message": "Invalid response from authorization hook"}]`. I created a new bug to track this issue: #1297 .

How we tested:
- Ran the unit test
- Manually verified instances of TinyInputField in the UI to ensure there were no styling regressions

Note: There are a lot of commits here, I'll squash them before merging to main.

### CODE REVIEW

**Comment by chelsea-tnc** [comment]
@pilar333 this is ready for review now, I synced it up with main to fix the merging conflicts.

File: script/build-newsletter-editions.js [hunk 1]
```diff
@@ -96,14 +96,16 @@ async function saveNewsletterEditions(organizationId, letterheadData) {
   for await (let newsletter of letterheadData.items) {
     let headline = shared.cleanContent(newsletter.title);
 
-    if (!newsletter.publicationDate) {
```
Comment (chelsea-tnc) [inline script/build-newsletter-editions.js:99]: I'm not sure why the left side shows the main branch with old code, the left side should match [what main already has](https://github.com/news-catalyst/next-tinynewsdemo/blob/main/script/build-newsletter-editions.js#L99).

File: script/build-newsletter-editions.js [hunk 2]
```diff
@@ -96,14 +96,16 @@ async function saveNewsletterEditions(organizationId, letterheadData) {
...
+    if (newsletter.publicationStatus != 1) {
       //from Letterhead: Publication status of results to include (1 = published; 2 = scheduled; 4 = review; 6 = all statuses)
       console.log(
         '> Org#' +
           organizationId +
           ' Newsletter ID#' +
           newsletter.id +
           " '" +
+          newsletter.publicationStatus +
```
Comment (chelsea-tnc) [inline script/build-newsletter-editions.js:107]: This looks identical to what [main](https://github.com/news-catalyst/next-tinynewsdemo/blob/main/script/build-newsletter-editions.js#L107) already has.

File: script/build-newsletter-editions.js [hunk 3]
```diff
@@ -173,6 +175,9 @@ async function saveNewsletterEditions(organizationId, letterheadData) {
           newsletter.title +
           "' was published at " +
           newsletter.publicationDate +
+          " '" +
```
Comment (chelsea-tnc) [inline script/build-newsletter-editions.js:178]: Same as above, looks [identical](https://github.com/news-catalyst/next-tinynewsdemo/blob/main/script/build-newsletter-editions.js#L178) to main.

**Comment by chelsea-tnc** [comment]
Conflicts resolved, this is ready for PR and merge.

**Comment by chelsea-tnc** [comment]
@pdeharo333 not sure if you tried this already, but I'd try running `npm install` first.

**Comment by pdeharo333** [comment]
That did work thanks! Sample test passed, and Hasura and queries test in failing as expected.

**Pull request merged**